Update solarmax-maxstorage.yaml

baloo-gh opened the pull request:
Solarprognose anhand der realen Produktionsdaten anpassen für das template solarmax-maxstorage.

Im Augenblick wird beim benutzen dieses templates die produzierte Energie nicht in EVCC eingeladen, daher ist dann die o.g. Korrektur nutzlos, da sie dann den Forecast um -100% auf 0 setzt. Für user wie mich, bei denen das Feature "connectivity+" freigeschaltet wurde, ist das schade, denn die Energie ist auslesbar, wird aber vom Template nicht ausgelesen.

Mit diesem zusätzlichen Code, der "energy" ausliest, sollte dies möglich sein.

baloo-gh committed:
Update solarmax-maxstorage.yaml

baloo-gh commented:
Ich hab es mit dem qModMaster ausprobiert - der Wert für Energie wird mit dem im code angegebenen Inputregister korrekt ausgelesen. 

Den code als solches konnte ich nicht testen (ich weiß nicht, wie ich das upgedatete template bei mir einsetzen kann in Home Assistant um auch das 100% zu validieren zu können.

andig commented:
https://github.com/evcc-io/evcc/blob/master/CONTRIBUTING.md#device-templates. Ohne Test können wir's nicht mergen.

baloo-gh committed:
Update solarmax-maxstorage.yaml

baloo-gh committed:
Merge branch 'master' into patch-2

baloo-gh committed:
Update solarmax-maxstorage.yaml

baloo-gh committed:
Update solarmax-maxstorage.yaml

This is proven to be working - tested in an EVCC environment with the real physical hardware

baloo-gh committed:
Merge branch 'master' into patch-2

andig commented:
> Verschieben in den PV Bereich hat nicht funktioniert

doch, tut es. Bitte ändern.

andig commented:
Da wo if pv steht

baloo-gh committed:
Update solarmax-maxstorage.yaml

andig commented:
Und jetzt noch die Batterie analog PV reparieren- da fehlt auch power

baloo-gh committed:
Update solarmax-maxstorage.yaml

premultiply commented:
Bitte füge noch die entsprechende Modbus-Dokumentation des Herstellers oder einer anderen validen Quelle bei auf das sich das hier bezieht bei.
Wichtig ist erkennen zu können das es hier nicht um irgendeine Summenenergie des Netzanschlusses eines Hybrid-Geräts sondern ganz explizit nur die erzeugte Energie.

baloo-gh commented:
Sehr gerne. Hier ist das offizielle Dokument von Solarmax:

[MAX.STORAGE.ModbusTCP.Datenpunktliste_extended.3.4.4.pdf](https://github.com/user-attachments/files/22948055/MAX.STORAGE.ModbusTCP.Datenpunktliste_extended.3.4.4.pdf)

Danke fürs mergen, schön, dass ich ein bisschen helfen konnte! Bin schon Sponsor bei Euch, nun auch Mini-Contributor :-)
